lib/stages/nDiskMultiFormatWriter.cpp

```diff
@@ -0,0 +1,392 @@
+    while (!stop_thread) {
+
+        // This call is blocking.
+        frame = wait_for_full_frame(buf, unique_name.c_str(), frame_id);
+        if (frame == nullptr)
+            break;
+
+        // INFO("Got buffer id: {:d}, disk id {:d}", frame_id, disk_id);
+
+        // Check if the producer has finished, and we should exit.
+        if (frame_id == -1) {
+            break;
+        }
+
+        // Open the file to write
+        if (write_to_disk) {
+
+            if (fd < 0) {
+                // make the output file name
+                file_name = get_dataset_folder_name(disk_id) + fmt::sprintf("/%010zu.", file_num)
+                            + file_extension;
+                // open a file and get its handle
+                fd = open(file_name.c_str(), O_WRONLY | O_CREAT, 0666);
+
+                // set num_frames to -1 just to write -1 at the beginning of the file
+                num_frames = -1;
+                // write the number of frames in this file
+                bytes_written = write(fd, &num_frames, sizeof(num_frames));
+                // reset the num_frames
+                num_frames = 0;
+            }
+
+            if (fd == -1) {
+                ERROR("Cannot open file");
+                ERROR("File name was: {:s}", file_name);
+                exit(errno);
+            }
+
+            // write the metadata and its size
+            size_t metadata_size = 0;
+            struct metadataContainer* mc = get_metadata_container(buf, frame_id);
+            if (mc != nullptr) {
+                metadata_size = mc->metadata_size;
+            }
+            // Write metadata size to disk, if there is no metadata in the frame, then
+            // just save 0 to the first word.
+            if (write(fd, (void*)&metadata_size, sizeof(metadata_size))
+                != (int32_t)sizeof(metadata_size)) {
+                ERROR("Failed to write metadata_size to disk for file {:s}", file_name);
+                exit(-1);
+            }
+            if (mc != nullptr) {
+                if (write(fd, mc->metadata, mc->metadata_size) != (int32_t)mc->metadata_size) {
+                    ERROR("Failed to write metadata_size to disk for file {:s}", file_name);
+                    exit(-1);
+                }
+            }
+
+            // write the number of frequencies
+            bytes_written =
+                write(fd, &num_freq_per_output_frame, sizeof(num_freq_per_output_frame));
+
+            // write the frame size in bytes
+            size_t frame_byte_size = (size_t)buf->frame_size;
+            bytes_written = write(fd, &frame_byte_size, sizeof(size_t));
+
+            // write the frame data
+            bytes_written = write(fd, frame, buf->frame_size);
+
+            if (bytes_written != buf->frame_size) {
+                ERROR("Failed to write buffer to disk!!!  Abort, Panic, etc.");
+                exit(-1);
+            } else {
+                // INFO("Data written to file!");
```

Comment:
Remove or turn into a `DEBUG2()` type statement.

Reply:
It seems unnecessary, so I removed it.